### ISSUE TITLE
clear all blocklist with fail2ban

### ISSUE THREAD

**Issue by doramir** [kickstarter/rack-attack#208]
Is there a way to release all everyone in the blocklist? 
this command  ` Rack::Attack.clear!` is not releasing  the blocklist.
I want to release everyone from blocklist

**Comment by alexvco** [comment]
Were you able to figure this one out?

**Comment by aishwarya923** [comment]
`Rack::Attack.clear!` does not release the users in the blocklist.
Rather, it clears the the configuration for throttle, blacklist, safelist and track.

Although I am not very sure how to release everyone from the blocklist, maybe clear from the cache?

**Comment by DCookieMonster** [comment]
I created a hack to solve it.
I used dynamic prefix for the key and when I wanted to clear  I just change the prefix key

**Comment by aishwarya923** [comment]
Similar issue also discussed here: https://github.com/kickstarter/rack-attack/issues/113

```
email = "user@example.com"
epoch_time = Time.now.to_i
Rack::Attack.cache.store.delete "#{Rack::Attack.cache.prefix}:#{epoch_time / 30.minutes}:logins/email:#{email}"
```

**Comment by DCookieMonster** [comment]
We don't want specific, I want to clear all the blacklist

**Comment by grzuy** [comment]
`Rack::Attack.clear!` was misleading and was changed to `Rack::Attack.clear_configuration` in https://github.com/kickstarter/rack-attack/pull/344.

**Comment by grzuy** [comment]
FWIW we're considering adding a `Rack::Attack.reset!` method to reset Rack::Attack's state in other issues (https://github.com/kickstarter/rack-attack/issues/389, https://github.com/kickstarter/rack-attack/issues/249#issuecomment-539029878).

**Comment by jtoy** [comment]
adding it?

**Comment by grzuy** [comment]
`Rack::Attack.reset!` was released in [`v6.3.0`](https://github.com/kickstarter/rack-attack/blob/master/CHANGELOG.md#630---2020-04-26). Be aware it resets every Rack::Attack state, no only fail2ban.

**Comment by sekmo** [comment]
When you say Rack::Attack state, are you talking about the cache and/or the rules?
In case, Is there a way to reset the rules? I'm doing some debugging on a staging console and I'm still searching for a way of redefining rules :)
